File: 2017/Day_01/code.php
<?php

require_once __DIR__ . '/../../lib/php/utils.php';

$input = readInput(__DIR__)[0];

// Task code
function part01(string $input): int
{
    $sum = 0;

    for ($i = 0, $totalCount = strlen($input); $i < $totalCount; $i++) {
        if ($input[$i] === $input[$i - 1]) {
            $sum += $input[$i];
        }
    }

    return $sum;
}

function part02(string $input): int
{
    $sum = 0;

    for ($i = 0, $totalCount = strlen($input); $i < $totalCount; $i++) {
        if ($input[$i] === $input[$i - ($totalCount/2)]) {
            $sum += $input[$i];
        }
    }

    return $sum;
}

// Execute
calcExecutionTime();
$result01 = part01($input);
$result02 = part02($input);
$executionTime = calcExecutionTime();

writeln('Solution Part 1: ' . $result01);
writeln('Solution Part 2: ' . $result02);
writeln('Execution time: ' . $executionTime);

saveBenchmarkTime($executionTime, __DIR__);

// Task test
testResults(
    [1223, 1284], // Expected
    [$result01, $result02], // Result
);

File: 2015/Day_04/code.go
package main

import (
	"advent-of-code/helper"
	"crypto/md5"
	"encoding/hex"
	"fmt"
	"path"
	"runtime"
	"strings"
	"time"
)

func main() {
	_, f, _, _ := runtime.Caller(0)
	cwd := path.Join(path.Dir(f))

	input := helper.ReadInput(cwd, helper.NewLine)
	iValues := input.Strings()

	// Execute
	start := time.Now()
	result01 := part01(iValues)
	result02 := part02(iValues)
	executionTime := helper.ExecutionTime(time.Since(start))

	fmt.Printf("Solution Part 1: %v\n", result01)
	fmt.Printf("Solution Part 2: %v\n", result02)
	fmt.Printf("Execution time: %s\n", executionTime)

	helper.SaveBenchmarkTime(executionTime, cwd)

	// Testing
	helper.TestResults(
		[]helper.TestingValue{
			helper.TestingValue{Result: result01, Expect: 254575},
			helper.TestingValue{Result: result02, Expect: 1038736},
		},
	)
}

// Task code
func part01(input []string) int {
	return findLowestPositiveNumber(input[0], "00000")
}

func part02(input []string) int {
	return findLowestPositiveNumber(input[0], "000000")
}

func findLowestPositiveNumber(key string, prefix string) int {
	counter := 0

	for {
		b := []byte(fmt.Sprintf("%s%d", key, counter))
		hash := md5.Sum(b)

		if strings.HasPrefix(hex.EncodeToString(hash[:]), prefix) {
			return counter
		}

		counter++
	}
}
